independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java

```diff
@@ -155,13 +157,14 @@ private void putContext(InjectableContext context) {
         if (values == null) {
             contexts.put(context.getScope(), Collections.singleton(context));
         } else {
-            List<InjectableContext> multi = new LinkedList<>(values);
+            List<InjectableContext> multi = new ArrayList<>(values.size() + 1);
```

Comment:
-1. This branch is only used if multiple contexts are registered for the same scope annotation, which is possible but rather rare in case of quarkus. Also what's wrong with a `LinkedList`? I seriously doubt that this change would cause a noticable difference...

Reply:
- https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/LinkedList

Reply:
LinkedLists are a very poor choice, especially if we care about the memory usage: you need an additional object to wrap each element in the list. And it just doesn't have any benefits.